bool[] w1
float32[] pos_x
float32[] pos_y
int32 ID
